client/network.go

```diff
@@ -56,7 +56,7 @@ func ConnectNetContext(ctx context.Context, routerURL string, cfg Config) (*Clie
 		routerURL = u.String()
 		fallthrough
 	case "ws", "wss":
-		p, err = transport.ConnectWebsocketPeerContext(ctx, routerURL, cfg.Serialization,
+		p, _, err = transport.ConnectWebsocketPeerContext(ctx, routerURL, cfg.Serialization,
```

Comment:
Seems like something should be done with the response here:
1. Handle redirect
2. Include HTTP status and response body, or whole response, in error

Consider a new type, `transport.WebSockerError` that can be returned as an `error`:
```
type WebsockerError struct {
    Err error
    Response *http.Response
}
func (e *WebsockerError) Error() string {
    if e.Response == nil {
        return e.Err.Error()
    }
    return fmt.Sprintf("%s: %s", e.Err, e.Response.Status)
}
```

This allows the caller to cast the error to retrieve the http response:
```
if wsErr, ok := err.(*transport.WebsocketError); ok {
    rsp := wsErr.Response
    // do something with rsp
}
```
This error should come from the transport code.